Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -0,0 +1,418 @@
+        throw
+    }
+}
+
+function Get-AzStorageAccount
+{
+    [CmdletBinding()]
+    param([String] [Parameter(Mandatory = $true)] $storageAccountName,
+          [Object] [Parameter(Mandatory = $true)] $endpoint)
+
+    try
+    {
+        $subscriptionId = $endpoint.Data.SubscriptionId
+        $azureUri = Get-AzureUri $endpoint
+
+        $uri="$azureUri/$subscriptionId/services/storageservices/$storageAccountName"
+        $headers = @{"x-ms-version"="2016-03-01"}
+        $method="GET"
+
+        $certificate = Get-Certificate $endpoint
+
+        $proxyUri = Get-ProxyUri $uri
+        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $uri))
+        {
+            $storageAccount=Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -Certificate $certificate
+            Write-Verbose "No Proxy settings"
+            return $storageAccount.StorageService.StorageServiceProperties
+        }
+        else
+        {
+            $storageAccount=Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -Certificate $certificate -UseDefaultCredentials -Proxy $proxyUri -ProxyUseDefaultCredentials
+            Write-Verbose "Using Proxy settings"
+            return $storageAccount.StorageService.StorageServiceProperties
+        }
+    }
+    catch
+    {
+        $exceptionMessage = $_.Exception.Message.ToString()
+        Write-Error "ExceptionMessage: $exceptionMessage"
+        throw
+    }
+}
+
+function Get-AzRmStorageAccount
+{
+    [CmdletBinding()]
+    param([String] [Parameter(Mandatory = $true)] $resourceGroupName,
+          [String] [Parameter(Mandatory = $true)] $storageAccountName,
+          [Object] [Parameter(Mandatory = $true)] $endpoint)
+
+    try
+    {
+        $accessToken = Get-SpnAccessToken $endpoint
+        $resourceGroupDetails = Get-AzRmResourceGroup $resourceGroupName $endpoint
+        $resourceGroupId = $resourceGroupDetails.id
+        $azureRmUri = Get-AzureUri $endpoint
+
+        $method="GET"
+        $uri = "$azureRmUri$resourceGroupId/providers/Microsoft.Storage/storageAccounts/$storageAccountName" + '?api-version=2016-01-01'
+
+        $headers = @{"x-ms-client-request-id"="a21c4b0a-2226-4ab5-a473-e39459e6369a"}
+        $headers.Add("Authorization", ("{0} {1}" -f $accessToken.token_type, $accessToken.access_token))
+
+        $storageAccountUnformatted = $null
+        $proxyUri = Get-ProxyUri $uri
+        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $uri))
+        {
+            $storageAccountUnformatted=Invoke-RestMethod -Uri $uri -Method $method -Headers $headers
+            Write-Verbose "No Proxy settings"
+        }
+        else
+        {
+            $storageAccountUnformatted=Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -UseDefaultCredentials -Proxy $proxyUri -ProxyUseDefaultCredentials
+            Write-Verbose "Using Proxy settings"
+        }
+
```

Comment:
Can you add a log message here, something like constructing the storage account.